public-ui/src/app/data-browser/charts/chart-sources/chart-sources-react.component.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+import { BaseReactWrapper } from 'app/data-browser/base-react/base-react.wrapper';
+import { getBaseOptions } from 'app/data-browser/charts/react-base-chart/base-chart.service';
+import * as highCharts from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+import * as React from 'react';
+
+interface State {
+    options: any;
+}
+
+interface Props {
+    concepts: any;
+}
+
+export class SourcesChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null};
+  }
+
+  componentDidMount() {
+      this.getChartOptions();
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    if (prevProps.concepts !== this.props.concepts) {
+        this.getChartOptions();
+    }
+  }
+
+  getChartOptions() {
+    const {concepts} = this.props;
+    const {categories, series} = this.prepCategoriesAndData(concepts);
+    this.setChartOptions(categories, series);
+  }
+
+  setChartOptions(categories: any, series: any) {
+      const newBaseOptions = getBaseOptions();
+      newBaseOptions.chart.type = 'column';
+      newBaseOptions.chart.tooltip =  {
+        headerFormat: '<span>{point.key} <br/>',
+        pointFormat: '{point.y}</span>'
+      };
+      newBaseOptions.plotOptions.column.pointPadding = 0.25;
+      newBaseOptions.plotOptions.series.minPointLength = 3;
+      newBaseOptions.plotOptions.column.pointWidth = 20;
+      newBaseOptions.xAxis.title.text = 'Source Concepts';
+      newBaseOptions.yAxis.title.text = 'Participant Count';
+      newBaseOptions.xAxis.categories = categories;
+      newBaseOptions.series = [series];
+      this.setState({options: newBaseOptions});
+  }
+
+  prepCategoriesAndData(concepts) {
+    const data = [];
+    const cats = [];
+    const COLUMN_COLOR = '#2691D0';
+    concepts = concepts.sort((a, b) => {
+      return a.sourceCountValue < b.sourceCountValue;
+    });
+    for (const a of concepts) {
+      let toolTipText = '';
+      let count;
+      count = (a.sourceCountValue <= 20) ? '&le; 20' : a.sourceCountValue;
+              toolTipText = '<div class="chart-tooltip">' + a.conceptName +
+              ' (' + a.vocabularyId + '-' + a.conceptCode + ') ' +
+              '<br/>' + 'Participant Count: ' + '<strong>' + count + '</strong> </div>';
+      data.push({
+        name: a.conceptName + ' (' + a.vocabularyId + '-' + a.conceptCode + ') ',
+        y: a.sourceCountValue, analysisId: 'sources',
+        color: COLUMN_COLOR,
+        toolTipHelpText: toolTipText
+      });
+      cats.push(a.vocabularyId + '-' + a.conceptCode);
+    }
+    const temp = data.filter(x => x.y > 20);
+    const dataOnlyLT20 = temp.length > 0 ? false : true;
```

Comment:
Can simplify the condition to `temp.length === 0` and optionally convert to a one-liner:
```suggestion
    const dataOnlyLT20 = data.filter(x => x.y > 20).length === 0;
```

Reply:
Changed this thanks.